org.eclipse.jdt.core.tests.builder/src/org/eclipse/jdt/core/tests/builder/DependencyTests.java

```diff
@@ -906,19 +909,20 @@ public void testMissingClassFile() throws JavaModelException {
 			"import p2.A;\n" +
 			"public class B {\n"+ //$NON-NLS-1$
 			"	public static void main(String[] args) {\n" + //$NON-NLS-1$
-			"		new A().foo(new String());\n" + //$NON-NLS-1$
+			"		new A().foo(new Object());\n" + // potentially ambiguous
+			"		new A().foo(new String());\n" + // exact match to fully resolved method
```

Comment:
I couldn't understand the "potentially ambiguous comment" here - should we just end up with the one method with the missing type - The test outcome is not questioned, just the comment

Reply:
sorry, "ambiguous" is wrong. I just tried to restore the original expected error now that `new A().foo(new String());` can be resolved without even looking at the other overload. Will change this to
`new A().foo(new B());\n" + 		// applicability test would like to see MissingClass`

This actually hints at further room for improvement (future): since the inheritance hierarchy of B is fully resolved, we _could_ rule out that it would ever be compatible to `MissingClass`. Not for now, though.